### PR TITLE
Add release-1.32 team and timeline

### PR DESCRIPTION
**Pull request by fsmunoz**
Initial v1.32 release information:

- directory
- release-team
- timeline

### CODE REVIEW

File: releases/release-1.32/README.md
Comment (katcosgrove) [inline releases/release-1.32/README.md:73]:
```suggestion
| Release Highlights deadline                                                                  | Comms                         | Tuesday 19th November 2024                                                    | week 11 |                                                        |
```

File: releases/release-1.32/README.md
Comment (katcosgrove) [inline releases/release-1.32/README.md:84]:
```suggestion
| Release Highlights complete                                                                  | Release Notes Lead            | Tuesday 3rd December 2024                                                     | week 13 |                                                        |
```

**Review by katcosgrove — COMMENTED** [review]
Major Themes is now Release Highlights, as of #2600 (yay!)

**Comment by fsmunoz** [comment]
Thank you @katcosgrove !

File: releases/release-1.32/README.md
Comment (liggitt) [inline releases/release-1.32/README.md:79]: During 1.31 cut we [discussed](https://kubernetes.slack.com/archives/C2C40FMNF/p1719340907948929?thread_ts=1719338931.466329&cid=C2C40FMNF) eliminating a distinct test freeze, and requiring stable test coverage be in place by code freeze. Fixes for test failures would still be accepted (just as they are today), but having a separate official test freeze date invites unstable or insufficient tests for code added before code freeze.

cc @johnbelamaric @BenTheElder @dims @thockin @Verolop @jeremyrickard @jpbetz @soltysh

Comment (liggitt) [reply releases/release-1.32/README.md:79]: I *think* that would just mean dropping this line and making the existing code freeze date the "code and test freeze"

File: releases/release-1.32/README.md
Comment (liggitt) [inline releases/release-1.32/README.md:75]: This is wednesday after kubecon. Speaking for myself, time for code reviews the previous week will be extremely scarce. If we put code freeze immediately after kubecon like this, we have to communicate super well that PRs not merged by start of kubecon could be ~high risk of missing, depending on who the reviewers / implementers are.

File: releases/release-1.32/README.md
Comment (BenTheElder) [reply releases/release-1.32/README.md:79]: cc @aojea @pohly 

+1, I think in retrospect test freeze has not been a helpful concept, and test fixes should be treated the same as bug fixes post-freeze, but whole new tests should not be deferred and ideally implementation PRs should have tests to begin with. 

Test-freeze encourages punting tests and we've even wound up with beta features that are not sufficiently tested due to deferred test work.

We could allow some additional time for code freeze into at least part of the test freeze window.

File: releases/release-1.32/README.md
Comment (alculquicondor) [reply releases/release-1.32/README.md:75]: I would rather move the code freeze before Kubecon, than give false hopes to contributors.

File: releases/release-1.32/README.md
Comment (jeremyrickard) [reply releases/release-1.32/README.md:79]: +1 to just dropping the line and rolling it into code freeze. @BenTheElder  is your suggestion to delay the code freeze period (shorten it?)

Comment (BenTheElder) [reply releases/release-1.32/README.md:79]: In the general case I think we should consider allowing code freeze to move to at least part of what was the test freeze period, but this time it might not make sense given https://github.com/kubernetes/sig-release/pull/2604#discussion_r1725396070

File: releases/release-1.32/README.md
Comment (jeremyrickard) [reply releases/release-1.32/README.md:75]: I would also vote to move this to before KubeCon. It also looks like we have docs deadlines during KubeCon. I would suggest both of these be week of Nov 4th. Perhaps also move the beta back? That would be a week between the alpha and the beta releases. I am not sure we need more than that based on how much feedback we typically get from them?

File: releases/release-1.32/README.md
Comment (BenTheElder) [reply releases/release-1.32/README.md:79]: We can either:
1. drop the test freeze date and roll it into the code freeze date
2. drop the code freeze date and roll it into what was the test freeze date
3. something in-between


Jordan had raised some concerns about 2), roughly that it means we have less time focused on stabilization.
3) might make sense, which is what I attempted to bring up, but for this release we also have https://github.com/kubernetes/sig-release/pull/2604#discussion_r1725396070 to consider (kubecon / freeze alignment ...)

File: releases/release-1.32/README.md
Comment (jeremyrickard) [inline releases/release-1.32/README.md:80]: This is also a holiday week in the US, so we should work early to make sure we have review coverage outside the US in terms of release managers.

File: releases/release-1.32/README.md
Comment (jeremyrickard) [reply releases/release-1.32/README.md:79]: I think for _most_ releases I would land on #3. I think for this release (like often happens with the end of the year), the calendar will work against us. 

So for *this* one, I would vote #1. 

For 1.33 I would vote #3.

File: releases/release-1.32/README.md
Comment (fsmunoz) [reply releases/release-1.32/README.md:75]: Thank you! Would something like this work:

-  Begin Feature blog freeze: Oct 30 (w8)
-  1.32.0-beta.0 release: Nov 5 (w9)
- Begin Code Freeze & Test Freeze: Nov 8 (w9)
- Docs deadline - PRs ready for review: Nov 18 (w11)

KubeCon NA is Nov 12-15, so this would anticipate `beta-0` 1 week, and put the freeze before KubeCon. Docs deadline could be on Nov 8 as well: I've suggested here to delay it one week instead of anticipating it.

The distance between `beta-0` and the freeze is shorter, and exception requests will appear on KubeCon week, but from what I've read here these are minor compared with the benefit of having the freeze before KubeCon.

File: releases/release-1.32/README.md
Comment (fsmunoz) [reply releases/release-1.32/README.md:80]: Thanksgiving week is 25-29, the problem I see with delaying things is that it could move the release date to week 15, which is closer to Christmas (also typically a period where vacations/holidays are a concern), but it's doable e.g.:

- Move release date to Dec 18th
- Move Docs Freeze to Dec 3rd
- Move `rc-1` to Dec 10 
 
... and adjust the rest, giving more days for the "ready to review"  ones. 

Or we could just keep it as is and prepare for it, as you mentioned @jeremyrickard .

Comment (fsmunoz) [reply releases/release-1.32/README.md:80]: Understood, will make sure that that is properly relayed to everyone.

File: releases/release-1.32/README.md
Comment (mbianchidev) [inline releases/release-1.32/README.md:76]: Release blog ready to review on the 19th could be hard for KubeCon reasons.

Shall we shift it of 1/2 days?
My proposal would be Thursday 21st or Friday 22nd.

File: releases/release-1.32/README.md
Comment (mbianchidev) [inline releases/release-1.32/README.md:77]: Proposal to shift feature blogs ready to review on Monday 24th of November.

Rationale is: 
- Avoid overlap of the Release Blog and the feature blogs ready-to-review deadlines
- Begin the ready-to-review ramble with a new PR wrangler for docs. My goal here will be to work closely with that PR wrangler (maybe even setting up a short check, async or sync for the overall blog statuses)

File: releases/release-1.32/README.md
Comment (mbianchidev) [inline releases/release-1.32/README.md:62]: On week 6 I would add a (soft) deadline for blog opt-ins.

The rationale behind this is to minimize the number of reachouts to KEP owners that are not interested in writing a blog but end up saying so only later in the process.

I would even add two questions in the template PR 
"Do you think that this feature requires a blog? If so, are you willing to write one?"

Just an idea, feel free to discard this if you think it's not feasible.

**Review by mbianchidev — CHANGES_REQUESTED** [review]

File: releases/release-1.32/README.md
Comment (fsmunoz) [reply releases/release-1.32/README.md:71]: 4th of November  IMO fits better with the other changes as well, using that.

File: releases/release-1.32/README.md
Comment (fsmunoz) [reply releases/release-1.32/README.md:76]: Friday 22nd looks doable, and helps for anyone travelling.

File: releases/release-1.32/README.md
Comment (fsmunoz) [reply releases/release-1.32/README.md:77]: Monday 25th November is the next usable day, changed.

File: releases/release-1.32/README.md
Comment (fsmunoz) [reply releases/release-1.32/README.md:62]: We've had that discussion (at least partially) in the[ 1.26 retro ](https://docs.google.com/document/d/11Oa-L76z8Y1wYufdm1wEV87NUJm1nzQ4b12k6UQY3t8)- I remember it because I suggested something like that:

> The deadlines for Feature Blogs are difficult to track and provide a status on: they are viewed as the day in which things need to be pushed, which makes it very hard to provide a status update since 80% of placeholder PRs arrive in the last 2 days. If we want to have realistic RAG (Red, Amber, Green) status for Comms, perhaps we should adopt a “soft” deadline and a “hard” one, or something else like a “rolling window” in which things get added at the end (we used this for 1 blog this release

The consensus was that "soft deadlines" do not exist, and every time they were used they were either perceived as "hard", or no deadline at all.

Interesting idea on adding the opt-in to the PR, would like to hear more opinions on it (I'm not sure if it's doable for this release from a practical perspective).

File: releases/release-1.32/README.md
Comment (fsmunoz) [reply releases/release-1.32/README.md]: I've committed a new version that reflects this approach, essentially compressing things a bit more to put deadlines before KubeCon, plus merging Code and Test Freeze.

File: releases/release-1.32/README.md
Comment (mbianchidev) [reply releases/release-1.32/README.md]: So my idea would be to have those questions directly here: https://github.com/kubernetes/kubernetes/blob/master/.github/ISSUE_TEMPLATE/enhancement.yaml

Since the enhancement tracking is done mainly for release purposes, adding a box with these questions seems reasonable. It would save an entire cycle of reachouts for whoever says directly "no, this enhancement won't have a blog" from the start. Wdyt?

I'm also ok with adding a hard deadline for the opt in on the 6th week but conscious that it probably won't be treated as such due to the nature of blogging/content creation in general.

File: releases/release-1.32/README.md
Comment (tallclair) [inline releases/release-1.32/README.md:43]:
```suggestion
- **02:00 UTC Friday 11th October 2024 / 19:00 PDT Thursday 10th October 2024**: Week 5 — [Enhancements Freeze](../release_phases.md#enhancements-freeze)
```

File: releases/release-1.32/README.md
Comment (saschagrunert) [inline releases/release-1.32/README.md:13]: ultra nit:
```suggestion
```

File: releases/release-1.32/README.md
Comment (saschagrunert) [inline releases/release-1.32/README.md:50]:
```suggestion
```

File: releases/release-1.32/README.md
Comment (soltysh) [reply releases/release-1.32/README.md]: :+1: for dropping the test freeze

File: releases/release-1.32/README.md
Comment (dipesh-rawat) [inline releases/release-1.32/README.md:43]: Minor suggestion _(optional)_: Consider adding KubeCon NA dates here for awareness, as we’ve done in the past.

```suggestion
- **Tuesday 12th - Friday 15th November 2024**: Week 10 - [KubeCon NA](https://events.linuxfoundation.org/kubecon-cloudnativecon-north-america/)
- **02:00 UTC Wednesday 20th November 2024 / 19:00 PDT Tuesday 19th November 2024**: Week 11 — [Code Freeze](../release_phases.md#code-freeze)
```

File: releases/release-1.32/README.md
Comment (dipesh-rawat) [inline releases/release-1.32/README.md:65]:
```suggestion
| 1.32.0-alpha.3 released                                                                | Branch Manager                | Tuesday 29th October 2024                                       | week 8   |                                                        |
```

File: releases/release-1.32/README.md
Comment (dipesh-rawat) [inline releases/release-1.32/README.md:71]: Minor suggestion _(optional)_: Consider adding KubeCon NA dates here for awareness, as we’ve done in the past.

```suggestion
| **KubeCon NA**                                                                         |                               | Tuesday 12th - Friday 15th November 2024                                    | week 10  |                                                        |
| **Begin [Burndown]** (Monday, Wednesday, and Friday meetings)                          | Lead                          | Monday 18th November 2024                                                   | week 11  |                                                        |
```

File: releases/release-1.32/README.md
Comment (dipesh-rawat) [inline releases/release-1.32/README.md:72]:
```suggestion
| Docs deadline — PRs ready for review                                                   | Docs Lead                     | Tuesday 19th November 2024                                                  | week 11  |                                                        |
```

File: releases/release-1.32/README.md
Comment (liggitt) [inline releases/release-1.32/README.md:43]: update with the fixed dates prior to kubecon

Comment (fsmunoz) [reply releases/release-1.32/README.md:43]: Agreed, thanks!

File: releases/release-1.32/README.md
Comment (fsmunoz) [reply releases/release-1.32/README.md:73]: Fixed in latest commit, thanks.

File: releases/release-1.32/README.md
Comment (fsmunoz) [reply releases/release-1.32/README.md]: Fixed in latest commit.

File: releases/release-1.32/README.md
Comment (fsmunoz) [reply releases/release-1.32/README.md]: I'm marking this as resolved based on the changes made to the timeline, which reflect the concerns shared here: if it doesn't, please feel free to reopen of course.

File: releases/release-1.32/README.md
Comment (katcosgrove) [reply releases/release-1.32/README.md]: We can't make unilateral changes to the KEP template. That is something that we would have to discuss with another SIG (so we can't do it this cycle), and it also doesn't really solve the problem; the templates aren't versioned, and people regularly miss updating to a new template with new questions. New templates aren't filled out for every enhancement every cycle, either. An owner may not want a blog for alpha, but does want one for GA. We are working on a separate solution for this.

For now, please ask about feature blogs as defined in your handbook.

**Comment by katcosgrove** [comment]
All comments and suggestions have been addressed. Let's get this show on the road!

/lgtm
/approve

File: releases/release-1.32/README.md
Comment (mbianchidev) [reply releases/release-1.32/README.md]: Got it, keen to know more about the separate solution!

**Pull request merged**
